engine/src/main/java/org/terasology/engine/logic/behavior/BehaviorComponent.java

```diff
@@ -2,16 +2,22 @@
 // SPDX-License-Identifier: Apache-2.0
 package org.terasology.engine.logic.behavior;
 
-import org.terasology.engine.entitySystem.Component;
 import org.terasology.engine.logic.behavior.asset.BehaviorTree;
+import org.terasology.gestalt.entitysystem.component.Component;
 import org.terasology.gestalt.module.sandbox.API;
 
 /**
  * Entities with this component are handled by a behavior tree. Default tree to fetch may be set.
  *
  */
 @API
-public class BehaviorComponent implements Component {
+public class BehaviorComponent implements Component<BehaviorComponent> {
     public BehaviorTree tree;
     public transient Interpreter interpreter;
+
+    @Override
+    public void copy(BehaviorComponent other) {
+        this.tree = other.tree;
```

Comment:
`BehaviorTree` is an asset - do we need to create deep copies of assets, or can we use the same instance?